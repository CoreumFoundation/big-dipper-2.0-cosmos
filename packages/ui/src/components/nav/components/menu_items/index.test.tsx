import renderer from 'react-test-renderer';
import MenuItems from '@/components/nav/components/menu_items';
import MockTheme from '@/tests/mocks/MockTheme';
import { RecoilRoot } from 'recoil';
// ==================================
// global setup
// ==================================
let component: renderer.ReactTestRenderer;

// ==================================
// mocks
// ==================================
const mockPush = jest.fn();

jest.mock('next/router', () => ({
  ...jest.requireActual('next/router'),
  useRouter: () => ({
    push: mockPush,
  }),
}));

// ==================================
// unit tests
// ==================================
describe('screen: Nav/MenuItems', () => {
  beforeEach(() => {
    component = renderer.create(
      <RecoilRoot>
        <MockTheme>
          <MenuItems />
        </MockTheme>
      </RecoilRoot>
    );
  });

  it('it renders', () => {
    const tree = component?.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
