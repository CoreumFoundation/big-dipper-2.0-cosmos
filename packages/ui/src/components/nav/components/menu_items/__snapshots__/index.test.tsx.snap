// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: Nav/MenuItems it renders 1`] = `
.emotion-0 {
  list-style: none;
  margin: 0;
  padding: 0;
  position: relative;
  padding-top: 8px;
  padding-bottom: 8px;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  min-width: 0;
  box-sizing: border-box;
  text-align: left;
  padding-top: 8px;
  padding-bottom: 8px;
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  padding-left: 16px;
  padding-right: 16px;
}

.emotion-1::-moz-focus-inner {
  border-style: none;
}

.emotion-1.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-1 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-1:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-1:hover {
    background-color: transparent;
  }
}

.emotion-1.Mui-selected {
  background-color: rgba(37, 214, 149, 0.08);
}

.emotion-1.Mui-selected.Mui-focusVisible {
  background-color: rgba(37, 214, 149, 0.2);
}

.emotion-1.Mui-selected:hover {
  background-color: rgba(37, 214, 149, 0.12);
}

@media (hover: none) {
  .emotion-1.Mui-selected:hover {
    background-color: rgba(37, 214, 149, 0.08);
  }
}

.emotion-1.Mui-focusVisible {
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-1.Mui-disabled {
  opacity: 0.38;
}

.emotion-1 .MuiListItemIcon-root {
  min-width: 48px;
}

.emotion-1.MuiListItem-gutters {
  padding: 16px 20px;
}

.emotion-1 .MuiListItemText-root {
  color: #999999;
}

.emotion-1.active {
  background: hsla(158, 70.5%, 73.8%, 0.2);
}

.emotion-1.active .MuiListItemIcon-root svg {
  fill: #25D695;
  color: #25D695;
}

.emotion-1.active .MuiListItemText-root {
  color: #25D695;
}

.emotion-2 {
  min-width: 56px;
  color: rgba(0, 0, 0, 0.54);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-3 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  min-width: 0;
  margin-top: 4px;
  margin-bottom: 4px;
}

.emotion-3.emotion-3 * {
  white-space: nowrap;
  text-overflow: ellipsis;
}

.emotion-4 {
  margin: 0;
  font-size: 1rem;
  white-space: pre-wrap;
  letter-spacing: 0.5px;
  font-family: "Noto Sans",sans-serif;
  font-weight: 400;
  line-height: 1.5;
  display: block;
}

.emotion-27 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-27:hover {
  cursor: pointer;
}

.emotion-27 svg {
  fill: #999999;
}

.emotion-27 svg path {
  fill: #999999;
}

.emotion-28 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  min-width: 0;
  margin-top: 4px;
  margin-bottom: 4px;
}

<ul
  className="MuiList-root MuiList-padding emotion-0"
>
  <a
    className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters emotion-1"
    href="/"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
  >
    <div
      className="MuiListItemIcon-root emotion-2"
    >
      <svg
        height={24}
        width={24}
      />
    </div>
    <div
      className="MuiListItemText-root emotion-3"
    >
      <span
        className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-4"
      >
        overview
      </span>
    </div>
  </a>
  <a
    className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters emotion-1"
    href="/blocks"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
  >
    <div
      className="MuiListItemIcon-root emotion-2"
    >
      <svg
        height={24}
        width={24}
      />
    </div>
    <div
      className="MuiListItemText-root emotion-3"
    >
      <span
        className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-4"
      >
        blocks
      </span>
    </div>
  </a>
  <a
    className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters emotion-1"
    href="/transactions"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
  >
    <div
      className="MuiListItemIcon-root emotion-2"
    >
      <svg
        height={24}
        width={24}
      />
    </div>
    <div
      className="MuiListItemText-root emotion-3"
    >
      <span
        className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-4"
      >
        transactions
      </span>
    </div>
  </a>
  <a
    className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters emotion-1"
    href="/validators"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
  >
    <div
      className="MuiListItemIcon-root emotion-2"
    >
      <svg
        height={24}
        width={24}
      />
    </div>
    <div
      className="MuiListItemText-root emotion-3"
    >
      <span
        className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-4"
      >
        validators
      </span>
    </div>
  </a>
  <a
    className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters emotion-1"
    href="/proposals"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
  >
    <div
      className="MuiListItemIcon-root emotion-2"
    >
      <svg
        height={24}
        width={24}
      />
    </div>
    <div
      className="MuiListItemText-root emotion-3"
    >
      <span
        className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-4"
      >
        proposals
      </span>
    </div>
  </a>
  <a
    className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters emotion-1"
    href="/params"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
  >
    <div
      className="MuiListItemIcon-root emotion-2"
    >
      <svg
        height={24}
        width={24}
      />
    </div>
    <div
      className="MuiListItemText-root emotion-3"
    >
      <span
        className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-4"
      >
        params
      </span>
    </div>
  </a>
  <div>
    <div
      className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters emotion-1"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiListItemIcon-root emotion-2"
      >
        <div
          className="emotion-27"
          role="button"
        >
          <svg />
        </div>
      </div>
      <div
        className="MuiListItemText-root emotion-28"
      >
        <span
          className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-4"
        >
          settings
        </span>
      </div>
    </div>
  </div>
</ul>
`;
